Sources/PubNub/Helpers/Crypto/CryptorModule.swift

```diff
@@ -0,0 +1,320 @@
+            .decryptionError,
+            additional: ["Decrypting resulted with an empty File"])
+          )
+        }
+        return .success($0)
+      }
+      .mapError {
+        PubNubError(.decryptionError, underlying: $0)
+      }
+    } catch let error as PubNubError {
+      return .failure(error)
+    } catch {
+      return .failure(PubNubError(
+        .decryptionError,
+        underlying: error,
+        additional: ["Could not decrypt InputStream"])
+      )
+    }
+  }
+  
+  private func cryptor(matching header: CryptorHeader) -> Cryptor? {
+    header.cryptorId() == defaultCryptor.id ? defaultCryptor : cryptors.first(where: {
+      $0.id == header.cryptorId()
+    })
+  }
+}
+
+/// Convenience methods for creating `CryptorModule`
+public extension CryptorModule {
+  
+  /// Returns **recommended** `CryptorModule` for encryption/decryption
+  ///
+  /// - Parameters:
+  ///   - key: Key used for encryption/decryption
+  ///   - withRandomIV: A flag describing whether random initialization vector should be used
+  ///
+  /// This method sets ``AESCBCCryptor`` as the primary object for decryption and encryption. It also instantiates ``LegacyCryptor`` with `withRandomIV`
+  /// flag in order to decode messages/files that were encoded in old way.
+  static func aesCbcCryptoModule(with key: String, withRandomIV: Bool = true) -> CryptorModule {
+    CryptorModule(default: AESCBCCryptor(key: key), cryptors: [LegacyCryptor(key: key, withRandomIV: withRandomIV)])
+  }
+  
+  /// Returns legacy `CryptorModule` for encryption/decryption
+  ///
+  /// - Parameters:
+  ///   - key: Key used for encryption/decryption
+  ///   - withRandomIV: A flag describing whether random initialization vector should be used
+  /// - Warning: It's highly recommended to always use ``aesCbcCryptoModule(with:withRandomIV:)``
+  static func legacyCryptoModule(with key: String, withRandomIV: Bool = true) -> CryptorModule {
+    CryptorModule(default: LegacyCryptor(key: key, withRandomIV: withRandomIV), cryptors: [AESCBCCryptor(key: key)])
+  }
+}
+
+extension CryptorModule: Equatable {
+  public static func ==(lhs: CryptorModule, rhs: CryptorModule) -> Bool {
+    lhs.cryptors.map { $0.id } == rhs.cryptors.map { $0.id }
+  }
+}
+
+extension CryptorModule: Hashable {
+  public func hash(into hasher: inout Hasher) {
+    hasher.combine(cryptors.map { $0.id })
+  }
+}
+
+extension CryptorModule: CustomStringConvertible {
+  public var description: String {
+    "Default cryptor: \(defaultCryptor.id), others: \(cryptors.map { $0.id })"
+  }
+}
+
+internal extension CryptorModule {
+  func encrypt(string: String) -> Result<Base64EncodedString, PubNubError> {
+    guard let data = string.data(using: defaultStringEncoding) else {
+      return .failure(PubNubError(
+        .encryptionError,
+        additional: ["Cannot create Data from provided String"]
+      ))
+    }
+    return encrypt(data: data).map {
+      $0.base64EncodedString()
+    }
+  }
```

Comment:
Did we have this in the past (I mean utility method)? 

Docs for Objective-C contains information about what type should be passed to encryption / decryption and how to create Base64 if resulting data should be sent further.

Reply:
Yes, I moved those methods from the `Crypto`. They are used internally by us in order to publish or retrieve encoded messages from History. [Here's](https://github.com/pubnub/swift/blob/master/Sources/PubNub/Helpers/Crypto/Crypto.swift#L215-L237) their current implementation in master.